### PR TITLE
XIVY-14998 add validation messages to rows

### PR DESCRIPTION
**Pull request by ivy-lgi**

### CODE REVIEW

**Comment by ivy-lgi** [comment]
validation

File: packages/dataclass-editor/src/master/ValidationRow.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import { MessageRow, SelectRow, TableCell } from '@axonivy/ui-components';
+import { flexRender, type Row } from '@tanstack/react-table';
+import { useAppContext } from '../context/AppContext';
+import type { DataClassField } from '../data/dataclass';
+import type { ValidationMessage } from '../protocol/types';
+import './ValidationRow.css';
+
+export const validationMessagesOfRow = (validationMessages: Array<ValidationMessage>, field: DataClassField) => {
+  return validationMessages.filter(message => message.path === field.name);
+};
```
Comment (ivy-lli) [inline packages/dataclass-editor/src/master/ValidationRow.tsx:10]: I suggest that also here you build an custom hook where you can get your validations for a specific path. Somethinkg like we've in the inscription client: https://github.com/axonivy/inscription-client/blob/master/packages/editor/src/context/useValidation.ts

File: packages/dataclass-editor/src/master/useValidation.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { useAppContext } from '../context/AppContext';
+import type { DataClassField } from '../data/dataclass';
+
+export const useValidation = (field: DataClassField) => {
```
Comment (ivy-lli) [inline packages/dataclass-editor/src/master/useValidation.ts:4]: I would directly add the name as param here and not the field. But just a personal opinion. This is fine too :)

File: packages/dataclass-editor/src/master/useValidation.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+  const { validationMessages } = useAppContext();
+  return validationMessages.filter(message => message.path === field.name);
```
Comment (ivy-lli) [inline packages/dataclass-editor/src/master/useValidation.ts:6]: maybe you can also think about to directly convert this validation messages to MessageData objects, which you then directly can use for the messageRows and also for the BasicFields...

**Review by ivy-lli — APPROVED** [review]

File: packages/dataclass-editor/src/master/useValidation.ts (hunk 3 above)
Comment (ivy-lgi) [reply packages/dataclass-editor/src/master/useValidation.ts:6]: Done in the next PR: https://github.com/axonivy/dataclass-editor-client/pull/45

**Pull request merged**